src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -46,9 +54,16 @@ public TElement[] ToArray()
 
             TElement[] array = new TElement[count];
             int[] map = SortedMap(buffer);
```

Comment:
Speaking of places where `try…fault` would be useful. `SortedMap` calls `Sort` on an `EnumerableSorter`. That calls `ComputeMap()` which you have changed to call `Rent`, and then `QuickSort` or `PartialQuickSort`, both of which can thow if the comparer throws. Since we don't have `try…fault` that should probably become:

```C#
internal int[] Sort(TElement[] elements, int count)
{
    int[] map = ComputeMap(elements, count);
    try
    {
            QuickSort(map, 0, count - 1);
    }
    catch
    {
        ArrayPool<int>.Shared.Return(map);
        throw;
    }
    return map;
}
```
And likewise for `PartialQuickSort`.